rest/tests/test_search.py

```diff
@@ -0,0 +1,303 @@
+                facet=urllib_parse.quote_plus(facet)
+            )
+        resp = self.client.get(
+            "{repo_base}{repo_slug}/search/?q={query}{facets}".format(
+                repo_base=REPO_BASE,
+                repo_slug=self.repo.slug,
+                query=urllib_parse.quote_plus(query),
+                facets=selected_facets_arg
+            )
+        )
+        self.assertEqual(HTTP_200_OK, resp.status_code)
+        return as_json(resp)
+
+    def assert_result_equal(self, result, resource):
+        """Helper method to assert result == resource."""
+        def make_text(obj):
+            """Format result.text"""
+            try:
+                # Strip XML tags from content before indexing.
+                tree = etree.fromstring(obj.content_xml)
+                content = etree.tostring(tree, encoding="utf-8", method="text")
+            except etree.XMLSyntaxError:
+                # For blank/invalid XML.
+                content = obj.content_xml
+            try:
+                content = content.decode('utf-8')
+            except AttributeError:
+                # For Python 3.
+                pass
+
+            return "{0} {1} {2}".format(
+                obj.title, obj.description, content
+            )
+
+        self.assertEqual(
+            {
+                'xa_avg_grade': resource.xa_avg_grade,
+                'course': resource.course.course_number,
+                'description': resource.description,
+                'description_path': resource.description_path,
+                'resource_type': resource.learning_resource_type.name,
+                'lid': resource.id,
+                'xa_nr_attempts': resource.xa_nr_attempts,
+                'xa_nr_views': resource.xa_nr_views,
+                'preview_url': get_preview_url(resource),
+                'run': resource.course.run,
+                'text': make_text(resource),
+                'title': resource.title
+            },
+            result
+        )
+
+    def test_methods(self):
+        """
+        Verify that write methods won't work
+        """
+        url = "{repo_base}{repo_slug}/search/?q={query}".format(
+            repo_base=REPO_BASE,
+            repo_slug=self.repo.slug,
+            query=""
+        )
+
+        self.assertEqual(
+            HTTP_405_METHOD_NOT_ALLOWED, self.client.post(url).status_code)
+        self.assertEqual(
+            HTTP_405_METHOD_NOT_ALLOWED, self.client.patch(url).status_code)
+        self.assertEqual(
+            HTTP_405_METHOD_NOT_ALLOWED, self.client.put(url).status_code)
+
+    def test_blank(self):
+        """
+        Verify that a blank search will retrieve all results.
+        """
+        no_q_url = "{repo_base}{repo_slug}/search/".format(
+            repo_base=REPO_BASE,
+            repo_slug=self.repo.slug,
+        )
+
+        count = self.get_results()['count']
+        self.assertEqual(1, count)
```

Comment:
I don't think that 1 is a good number to test "all results". You should have at least 2